packages/dev/inspector-v2/src/components/properties/materials/normalMapProperties.tsx

```diff
@@ -3,10 +3,19 @@ import type { FunctionComponent } from "react";
 import { BoundProperty } from "../boundProperty";
 import { SwitchPropertyLine } from "shared-ui-components/fluent/hoc/propertyLines/switchPropertyLine";
 
-export type MaterialWithNormalMaps = {
-    invertNormalMapX: boolean;
-    invertNormalMapY: boolean;
-};
+export type MaterialWithNormalMaps =
+    | {
+          invertNormalMapX: boolean;
+          invertNormalMapY: boolean;
+      }
+    | {
+          _invertNormalMapX: boolean;
+          _invertNormalMapY: boolean;
+      };
+
+function CheckNormalMapPropertyNames(mat: MaterialWithNormalMaps): mat is { invertNormalMapX: boolean; invertNormalMapY: boolean } {
+    return (mat as any).invertNormalMapX !== undefined;
+}

```

Comment:
We can avoid the `any` cast:
```suggestion
type MaterialWithPublicNormalMaps = {
    invertNormalMapX: boolean;
    invertNormalMapY: boolean;
};

type MaterialWithInternalNormalMaps = {
    _invertNormalMapX: boolean;
    _invertNormalMapY: boolean;
};

export type MaterialWithNormalMaps = MaterialWithPublicNormalMaps | MaterialWithInternalNormalMaps;

function IsMaterialWithPublicNormalMaps(mat: MaterialWithNormalMaps): mat is MaterialWithPublicNormalMaps {
    return (mat as MaterialWithPublicNormalMaps).invertNormalMapX !== undefined;
}
```